utils/platform/platform_interface.go

```diff
@@ -0,0 +1,100 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package platform
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/imdario/mergo"
+	"github.com/sealerio/sealer/common"
+	v1 "github.com/sealerio/sealer/types/api/v1"
+	v2 "github.com/sealerio/sealer/types/api/v2"
+	utilsnet "github.com/sealerio/sealer/utils/net"
+	"github.com/sealerio/sealer/utils/os/fs"
+	"github.com/sirupsen/logrus"
+)
+
+type Interface interface {
+	// Platform Get remote platform
+	Platform(host net.IP) (v1.Platform, error)
+}
+
+func GetClusterPlatform(cluster *v2.Cluster) (map[string]v1.Platform, error) {
```

Comment:
One suggestion： this file only keeps interface definitions. looks like `GetClusterPlatform` is a upstream service to calls this interface.